solutions/USACO open16-closing-the-farm.mdx

```diff
@@ -2,16 +2,98 @@
 id: usaco-644
 source: USACO Silver 2016 Open
 title: Closing the Farm
-author: Maggie Liu
+author: Jesse Choe, Maggie Liu
 ---
 
 [Official Analysis](http://www.usaco.org/current/data/sol_closing_silver_open16.html)
 
+In this problem, we are asked to determine if all remaining barns are connected as the barns close one at a time. To check this, we can run a depth-first search (DFS) starting from the barn that will close last. To simulate the closure of each barn, we can store a boolean array and set each barn in the iteration to be "closed". Then, we can run the DFS and check how many nodes have been visited. If all the unclosed nodes have been visited, then we print "YES"; otherwise, we print "NO".
+
 <LanguageSection>
 
 <CPPSection>
 
-## C++ Implementation
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+using ll = long long;
+using vl = vector<ll>;
+using vb = vector<bool>;
+using pl = pair<ll, ll>;
+using vpl = vector<pl>;
+using vc = vector<char>;
+using vs = vector<string>;
+using str = string;
+
+#define FastIO ios_base::sync_with_stdio(0); cin.tie(0)
+#define pb push_back
+#define mp make_pair
+#define all(x) begin(x), end(x)
+#define sor(x) sort(all(x))
+#define ins insert
+#define f first
+#define s second
+#define sz(x) (ll)(x).size()
+#define len(x) (ll)(x).length()
+#define FOR(i, a, b) for(ll i=a; i<b; ++i)
```

Comment:
Why do you use 'long long' instead of 'int'?

Reply:
Should we ask benq?

Reply:
I transitioned from int to long long all together cuz I hate when u get like WA for forgetting long longs. I don't wanna go throught the hustle of changing, so I'm using ll instead of int @tachithanhdanh

Reply:
> I transitioned from int to long long all together cuz I hate when u get like WA for forgetting long longs. I don't wanna go throught the hustle of changing, so I'm using ll instead of int @tachithanhdanh

But is it necessary to convert the variables in the loop to `long long`?
Also, why do you have to use `long long` for `sz(x)` and `len(x)`, both of which will never exceed the size of `int`?

Reply:
The problem is that ints do have a significant advantage over lls, usually a noticeable amount of memory & runtime. Being able to automatically recognize when it's necessary to ll's over ints is a good skill too (and also lets you recognize when int128/bigint is necessary). So asking you to fix this would potentially fall under "bad practice," but we should probably ask @bqi343 if it's reasonable enough.

Reply:
and one thing to note is that `long long` variables tend to run much slower than that of `int` because they use twice as much memory as the `int` do.

Reply:
Wait, so how significant is the slow runtime for using `long long`?

Reply:
By almost a factor of 2. (Possibly more, given that the skipped soln only ran for pretests)

https://codeforces.com/contest/1348/submission/78738103
https://codeforces.com/contest/1348/submission/78739532